### PR TITLE
Update [Drawer] component (docs & examples)

### PR DESCRIPTION
**Pull request by manojleaton**
Fixes #BLUI-4365 .

#### Changes proposed in this Pull Request:
- Updated Drawer component and it's children components with MD3Theme.
- Please verify this PR for kitchen sink updates for Drawer component [DrawerKitchenSink](https://github.com/etn-ccis/blui-react-native-showcase-demo/pull/95)

#### Screenshots / Screen Recording (if applicable)
Screenshot 2023-10-25 at 11 58 58 AM

#### To Test:
- yarn start

#### Any specific feedback you are looking for?
-

### CODE REVIEW

**Review by ektaghag-eaton — COMMENTED** [review]
Drawer Header 

Title: 
    - LineHeight should be 32 px (need to fix in theme story just for a record adding comment here)
Subtitle: 
    - As per design it should use `bodyMedium` not `titleMedium`.
    - color should be #44474E need to update color code.

**Comment by ektaghag-eaton** [comment]
Please resolve conflicts.

**Comment by ArshdeepSingh3-eaton** [comment]
The above padding  is given to cover the status bar and is dynamically adjusted based on height of status bar

**Comment by ektaghag-eaton** [comment]
> The above padding is given to cover the status bar and is dynamically adjusted based on height of status bar

Thanks for explaining !!!

**Comment by ArshdeepSingh3-eaton** [comment]
Screenshot 2023-10-27 at 9 56 39 AM

**Comment by ektaghag-eaton** [comment]
@manojleaton and @ArshdeepSingh3-eaton please check comment https://github.com/etn-ccis/blui-react-native-showcase-demo/pull/95#pullrequestreview-1699181871

**Comment by ektaghag-eaton** [comment]
1. Title - Needs to be Eaton Font (need to add this in theme story for record adding here)
2. Subtitle - color and font are not as per the design
3. Active item text - font should be open-sans
4. The inactive Item's chevron color should be #44474E
5. Can you add nested items like below to check the background color of nested Items?
Screenshot 2023-10-27 at 1 44 54 PM
6. Drawer background color is #F2F6F8 i.e. `ui/surface-container-low`.

Following this https://www.figma.com/file/Nydz8n8HBj6U3ZtULja3gD/%F0%9F%92%AB-Brightlayer-UI-Tokens-%2B-Components-(in-Material-Design-3)?type=design&node-id=428-4719&mode=design&t=KNrFb82crFrKxeoN-0

**Comment by ArshdeepSingh3-eaton** [comment]
Screenshot 2023-10-27 at 2 13 03 PM

File: components/src/md3/Drawer/DrawerNavItem.tsx [hunk 1]
```diff
@@ -123,14 +123,13 @@ const makeStyles = (
...
 }> => {
-
     const {
         // Shared style props
         activeItemBackgroundColor = theme.colors.primaryContainer,
         activeItemBackgroundShape = 'square',
         backgroundColor,
         depth,
-        nestedBackgroundColor = theme.dark ? Colors.darkBlack[100] : theme.colors.background, // TODO: don't hardcode?
+        nestedBackgroundColor = theme.dark ? Colors.darkBlack[100] : theme.colors.surfaceContainer, // TODO: don't hardcode?
```
Comment (ektaghag-eaton) [inline components/src/md3/Drawer/DrawerNavItem.tsx:132]: We need to remove this reference of Colors.darkBlack[100].

**Comment by ektaghag-eaton** [comment]
> 1. Title - Needs to be Eaton Font (need to add this in theme story for record adding here)
> 2. Subtitle - color and font are not as per the design
> 3. Active item text - font should be open-sans
> 4. The inactive Item's chevron color should be #44474E
> 5. Can you add nested items like below to check the background color of nested Items?
> 
> Screenshot 2023-10-27 at 1 44 54 PM 6. Drawer background color is #F2F6F8 i.e. `ui/surface-container-low`.
> Following this https://www.figma.com/file/Nydz8n8HBj6U3ZtULja3gD/%F0%9F%92%AB-Brightlayer-UI-Tokens-%2B-Components-(in-Material-Design-3)?type=design&node-id=428-4719&mode=design&t=KNrFb82crFrKxeoN-0

2. Subtitle - color and font are not as per the design (more precise font-size is not matching)
3. Active item text - font should be open-sans

not fixed ..

**Review by ektaghag-eaton — APPROVED** [review]

**Pull request merged**